inference-engine/src/inference_engine/src/ie_network_reader.cpp

```diff
@@ -275,14 +286,26 @@ CNNNetwork details::ReadNetwork(const std::string& model,
             return reader->read(modelStream, exts);
         }
     }
+
     // Try to load with FrontEndManager
-    // NOTE: weights argument is ignored
-    const auto manager = get_frontend_manager();
+    auto& manager = get_frontend_manager();
     ngraph::frontend::FrontEnd::Ptr FE;
     ngraph::frontend::InputModel::Ptr inputModel;
-    FE = manager->load_by_model(&modelStream);
+
+    ov::VariantVector params{ov::make_variant(&modelStream)};
+    if (weights) {
+        auto weights_const = std::make_shared<ngraph::op::Constant>(ngraph::element::u8,
```

Comment:
Avoid data copy

Reply:
You can use SharedBuffer to avoid the copy

Reply:
Unfortunately SharedBuffer takes char * for data but weights has Blob::CPtr type.